server/models/Transaction.js

```diff
@@ -411,19 +411,54 @@ export default (Sequelize, DataTypes) => {
       transaction.amountInHostCurrency = Math.round(transaction.amountInHostCurrency);
     }
 
-    const oppositeTransaction = {
-      ...transaction,
-      type: -transaction.amount > 0 ? TransactionTypes.CREDIT : TransactionTypes.DEBIT,
-      FromCollectiveId: transaction.CollectiveId,
-      CollectiveId: transaction.FromCollectiveId,
-      HostCollectiveId: await models.Collective.getHostCollectiveId(transaction.FromCollectiveId), // see https://github.com/opencollective/opencollective/issues/1154
-      amount: -transaction.netAmountInCollectiveCurrency,
-      netAmountInCollectiveCurrency: -transaction.amount,
-      amountInHostCurrency: Math.round(-transaction.netAmountInCollectiveCurrency * transaction.hostCurrencyFxRate),
-      hostFeeInHostCurrency: transaction.hostFeeInHostCurrency,
-      platformFeeInHostCurrency: transaction.platformFeeInHostCurrency,
-      paymentProcessorFeeInHostCurrency: transaction.paymentProcessorFeeInHostCurrency,
-    };
+    // Is the target "collective" (account) "Active" (has an host, manage its own budget)
+    const fromCollective = await models.Collective.findByPk(transaction.FromCollectiveId);
+
+    let oppositeTransaction;
+    if (!fromCollective.isActive) {
+      oppositeTransaction = {
+        ...transaction,
+        type: -transaction.amount > 0 ? TransactionTypes.CREDIT : TransactionTypes.DEBIT,
+        FromCollectiveId: transaction.CollectiveId,
+        CollectiveId: transaction.FromCollectiveId,
+
+        HostCollectiveId: null,
+
+        amount: -transaction.netAmountInCollectiveCurrency,
+        netAmountInCollectiveCurrency: -transaction.amount,
+
+        amountInHostCurrency: Math.round(-transaction.netAmountInCollectiveCurrency * transaction.hostCurrencyFxRate),
+        hostFeeInHostCurrency: transaction.hostFeeInHostCurrency,
+        platformFeeInHostCurrency: transaction.platformFeeInHostCurrency,
+        paymentProcessorFeeInHostCurrency: transaction.paymentProcessorFeeInHostCurrency,
+      };
+    } else {
+      const oppositeTransactionCurrencyFxRate = await getFxRate(
+        transaction.currency,
+        fromCollective.currency,
+        transaction.createdAt,
+      );
+      oppositeTransaction = {
+        ...transaction,
+        type: -transaction.amount > 0 ? TransactionTypes.CREDIT : TransactionTypes.DEBIT,
+        FromCollectiveId: transaction.CollectiveId,
+        CollectiveId: transaction.FromCollectiveId,
+
+        // Added
+        HostCollectiveId: fromCollective.HostCollectiveId,
+
+        // Converted to fromCollective currency
+        currency: fromCollective.currency,
+        amount: -Math.round(transaction.netAmountInCollectiveCurrency * oppositeTransactionCurrencyFxRate),
+        netAmountInCollectiveCurrency: -Math.round(transaction.amount * oppositeTransactionCurrencyFxRate),
+        data: { oppositeTransactionCurrencyFxRate },
```

Comment:
Shouldn't we also persist the existing transaction data?
```suggestion
        data: { ...transaction.data, oppositeTransactionCurrencyFxRate },
```

Reply:
I considered, but I thought it was already persisted in the original transaction.

Reply:
I'm worried about being able to fetch CREDITs or DEBITs where `data.isFeesOnTop` is true in a simple query.
I'm not sure if there's any other case where we would be interested in fetching by data.

Reply:
Sorry, got it wrong, we should definitely do that.